### ISSUE TITLE
Making Q&A from the mattermost channel

### ISSUE THREAD

**Issue by vlukashenko** [lhcb/starterkit-lessons#224]
Maybe it would be useful to add some Q&A from the starterkit mattermost channel questions? 

p.s. I am happy to volunteer for maintaining it.

**Comment by alexpearce** [comment]
We've had, and still have, a long-standing problem of finding information 😞  If you have a question, where do you look to find the answer? Mailing lists, TWiki pages, the Starterkit, Mattermost, Doxygen pages, source code, that Discourse thing, and many more.

A Q&A would be another source of information. It's one more place to look, and one more place for the information to become outdated.

Having said all that, people do ask the same questions fairly often, and it would be great to have a (known) place to point them to for the answer.

Ideally the answers will be embedded in the lessons themselves. But having lots of 'trivia' throughout a lesson can make them hard to follow, disrupting the flow, so factoring out this trivia in to a Q&A page makes sense.

As you can tell, I'm somewhat on the fence about it 😄 Let's say a solid +0.5.

Am keen to hear what others think.

**Comment by chrisburr** [comment]
Thanks @alexpearce for saving me from writing something similar. 😆 (you did a better job!)

The discourse is available at https://lhcb-talk.web.cern.ch/ and has been advertised a few times but hasn't yet taken off yet. I also think it should be made public so the answers can start showing up on Google.

I think it's a good idea to copy some answered questions to `lhcb-talk` so they're more discoverable. This also doesn't have to be limited to the starterkit channel and could also include other channels and some mailing list questions. I keep meaning to encourage people to do this so if @vlukashenko is willing to promote it then it's a +1 from me.

**Comment by vlukashenko** [comment]
Thanks, @alexpearce and @chrisburr 
> A Q&A would be another source of information. It's one more place to look, and one more place for the information to become outdated.
Yes, I agree it's an issue. But I think that maybe a curated list would be a bit better solution. Now you have to search through many answers in mailing lists or mattermost until you find something useful (and it's the case if you know where and what to look for!). 

@chrisburr I think Daniel Campora mentioned https://lhcb-talk.web.cern.ch/ a few days ago to me and it was the first time I have heard of it. If we are talking about it now maybe it should be highlighted in the starterkit more actively?
However, I must say that due to the StackExchange-like nature of the lhcb-talk, it is indeed not easy to keep the answers up-to-date. It's a bit of an issue with the format itself. The same probably is true for the Q&A list too. Do you or @alexpearce have any idea what alternative formats can be used? 

> This also doesn't have to be limited to the starterkit channel and could also include other channels and some mailing list questions
I agree with my both hands with this statement. It shouldn't be constrained within the starterkit channels or mattermost channels.

**Comment by chrisburr** [comment]
> If we are talking about it now maybe it should be highlighted in the starterkit more actively?

It currently isn't as it didn't exist when the last starterkit happened :) (older messages from "system" were imported from an older webpage)

> However, I must say that due to the StackExchange-like nature of the lhcb-talk, it is indeed not easy to keep the answers up-to-date. It's a bit of an issue with the format itself.

I think it's not so bad. Ideally if someone finds an answer which is wrong, they should add a new answer correcting it. People searching for information can then learn to read backwards from the most recent reply. Of course that requires a "culture" of correcting documentation but I hope the format provides sufficiently little friction that it might happen. Especially if it can be encouraged in the starterkit or when people's questions are answered on mattermost.

**Comment by vlukashenko** [comment]
> It currently isn't as it didn't exist when the last starterkit happened :) (older messages from "system" were imported from an older webpage)
Oh, the system messages are a bit misleading indeed 😅 

> Ideally if someone finds an answer which is wrong, they should add a new answer correcting it.
In this case, indeed that would work. Maybe it will be also reasonable to introduce then tagging for which Run/year the answer is relevant? Idk if that is possible.

**Comment by chrisburr** [comment]
>  Maybe it will be also reasonable to introduce then tagging for which Run/year the answer is relevant?

Apparently it is, if they're not enabled we can ask Joel: https://meta.discourse.org/t/a-comprehensive-guide-to-discourse-tags/121041

**Comment by vlukashenko** [comment]
> Apparently it is, if they're not enabled we can ask Joel: https://meta.discourse.org/t/a-comprehensive-guide-to-discourse-tags/121041

I meant tagging the answer to the topic. I have never seen it in any of the similar forums, so I guess it's not really a feature. 

Anyway, should I then start copying certain question/answer pairs to the `lhcb-talk`?

**Comment by chrisburr** [comment]
From my perspective go ahead (the content can always be copied elsewhere if needed).

If anyone thinks we should try a different approach, speak up! 😄

**Comment by eduardo-rodrigues** [comment]
Hi @vlukashenko, all, I see a lot of enthusiasm. I wanted to remind you of the LHCb glossary that I started some time ago, which can be seen as a kind of scoped Q&A for definitions. I hope you can keep the glossary in mind during your endeavour, and even contribute to it or get others to do that.

The glossary https://lhcb.github.io/glossary/ is linked from the main LHCb page and as seen from the "issues" (https://github.com/lhcb/glossary/issues) many updates are pending ...

Thanks.

**Comment by vlukashenko** [comment]
@chrisburr Thanks for the green light ;)

@eduardo-rodrigues Sure, I totally forgot about the glossary at this point. Will use it for the definitions. (Probably will go through the list of the issues at the end of this week, maybe I can already give some nice definitions). Thanks!

**Comment by eduardo-rodrigues** [comment]
Brillant, that's very much appreciated @vlukashenko!
